File: Easy/roman_to_int.py
#Max Hillyer 4/27/25

class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        s += " "
        hash = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,
            " " : 0
        }
        num = 0

        for i in range(len(s) - 1):
            if hash.get(s[i]) < hash.get(s[i+1]):
                num -= hash.get(s[i])
            else:
                num += hash.get(s[i])
        return(num)

        

File: Easy/roman_to_int.c
//Max hillyer 6/28/25

int value(char c) {
    switch(c) {
        case 'I': return 1;
        case 'V': return 5;
        case 'X': return 10;
        case 'L': return 50;
        case 'C': return 100;
        case 'D': return 500;
        case 'M': return 1000;
        default: return 0;
    }
}

int romanToInt(char* s) {
    int total = 0;
    int i = 0;

    while (s[i]) {
        int s1 = value(s[i]);
        int s2 = value(s[i+1]);

        if (s2 > s1) {
            total += s2 - s1;
            i += 2;
        } else {
            total += s1;
            i += 1;
        }
    }

    return total;
}

